lib/using_restricted_array.rb

```diff
@@ -1,61 +1,117 @@
-require_relative 'restricted_array.rb'
+require_relative "restricted_array.rb"
 # RestrictedArray can be created using a specified size, or a random size in
 # the range of 1-20 will be chosen for you.
 # All values are integers in the range of 1-221.
 # RestrictedArray cannot be resized.
 
 # Calculates the length of the restricted array. All values are integers.
 # The restricted_array is terminated by 'nil' i.e. array[length] = nil
-# Time complexity: ?
+# Time complexity: Linear or O(n), where n is the number of items in the array or the length of the input #array
 # Space complexity: ?
 def length(array)
-  raise NotImplementedError
+  i = 0
+  # until array[i] == nil
+  while array[i] != nil
+    i += 1
+  end
+  return i
 end
 
 # Prints each integer values in the array
-# Time complexity: ?
+# Time complexity: linear or O(n),where n is the number of items in the array
 # Space complexity: ?
 def print_array(array)
-  raise NotImplementedError
+  # [3, 7, 9]
+  len = length(array)
+  i = 0
+  while i < len
+    print array[i]
+    print " "
+    i += 1
+  end
+  puts
 end
 
 # For an unsorted array, searches for 'value_to_find'.
 # Returns true if found, false otherwise.
-# Time complexity: ?
+# Time complexity: O(n), where n is the number of values in the array
 # Space complexity: ?
 def search(array, length, value_to_find)
-  raise NotImplementedError
+  length.times do |i|
+    return true if array[i] == value_to_find
+  end
+  return false
 end
 
 # Finds and returns the largest integer value the array
 # Assumes that the array is not sorted.
-# Time complexity: ?
+# Time complexity: O(n), where n is the number of values in the array
 # Space complexity: ?
 def find_largest(array, length)
-  raise NotImplementedError
+  largest = 0
```

Comment:
Setting largest = 0 works if your input array has all positive numbers. But if the input array is [-1, -2, -5] your fn would return 0 not -1.